larray/io/excel.py

```diff
@@ -321,10 +321,9 @@ def __setitem__(self, key, value):
         @property
         def shape(self):
             # include top-left empty rows/columns
-            # XXX: is there an exposed xlwings API for this? expand maybe?
-            used = self.xw_sheet.api.UsedRange
-            return (used.Row + used.Rows.Count - 1,
-                    used.Column + used.Columns.Count - 1)
+            # XXX: requires xlwings 0.9+
+            used = Range(self, 'A1').xw_range.current_region
```

Comment:
I wanted to avoid this because it will not take all data if you have a blank column/row in your sheet. That's why I used "usedRange" (which is equivalent to Ctrl-End in Excel). See the "bad2" sheet in the new file I uploaded on the issue. BTW: "bad2" is loaded "correctly" (as I think we should) by the xlrd engine. On the other hand having the behavior your just implemented is often what user want, so if we can provide an option for that, that would be nice. I would still include everything (except trailing blanks) by default, because it is easier/more intuitive to take a subset when you received too much rather than just miss some data (possibly without noticing).